client/my-sites/email/email-management/home/email-mailbox-warnings.jsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@automattic/components';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS,
+	EMAIL_WARNING_SLUG_UNVERIFIED_FORWARDS,
+} from 'calypso/lib/emails/email-provider-constants';
+import {
+	getEmailForwardAddress,
+	hasGoogleAccountTOSWarning,
+	isGoogleEmailAccount,
+} from 'calypso/lib/emails';
+import { getGmailUrl } from 'calypso/lib/gsuite';
+import Gridicon from 'calypso/components/gridicon';
+import { isEmailForwardAccount } from 'calypso/lib/emails/is-email-forward-account';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { resendVerificationEmail } from 'calypso/state/email-forwarding/actions';
+
+const EmailMailboxWarningText = ( { text } ) => {
+	return (
+		<div className="email-mailbox-warnings__warning">
+			<Gridicon icon="info-outline" size={ 18 } />
+			<span>{ text }</span>
+		</div>
+	);
+};
+
+const EmailMailboxWarningAction = ( { buttonText, isExternal, ...otherProps } ) => {
```

Comment:
It may be worth it to document this component directly, or decorate it with PropTypes. I worry that it will be difficult for third parties to use `otherProps` without referring to the component source.

Reply:
I'd be happy to add `PropTypes` for the directly supported properties, but the best I can do for `otherProps` is to reference the props/documentation for `Button` anyway.